server/src/main/java/org/elasticsearch/cluster/metadata/MetadataCreateIndexService.java

```diff
@@ -1411,6 +1411,17 @@ private static void updateIndexMappingsAndBuildSortOrder(
         }
     }
 
+    private static void validateSourceMode(MapperService mapperService) {
+        final IndexSettings indexSettings = mapperService.getIndexSettings();
+        if (indexSettings != null
+            && indexSettings.getIndexVersionCreated().onOrAfter(IndexVersions.DEPRECATE_SOURCE_MODE_MAPPER)
+            && SourceFieldMapper.isSynthetic(indexSettings) == false
+            && mapperService.documentMapper() != null
+            && mapperService.documentMapper().sourceMapper().isSynthetic()) {
```

Comment:
If we check whether `index.mapping.source.mode` has not be defined, then I think ` mapperService.documentMapper().sourceMapper().isSynthetic()` can return true if index.mode=logsdb|time_series and no _source.mode has been defined?